[FIX] deploy: set Content-Security-Policy on static (#5485)
odoo/documentation: content/administration/install/deploy.rst

by intercepting all requests to :samp:`/{MODULE}/static/{FILE}`, and looking up the right module
(and file) in the various addons paths.

It is recommended to set the ``Content-Security-Policy: default-src 'none'`` header on all images
delivered by the web server. It is not stricly necessary as users cannot modify/inject content
inside of modules' :file:`static/` folder and that existing images are final: they don't fetch new
resources by themselves. Still, it is good hygiene.

> Using the above NGINX (https) configuration, the following ``map`` and ``location`` blocks should be

Reviewer: > Using the above NGINX (https) configuration,

In this case the word bellow (ci-dessous) would be more appropriate

```
Using the NGINX (https) configuration provided bellow
```

> Using the above NGINX (https) configuration, the following ``map`` and ``location`` blocks should be

Author: I'm actually referencing the NGINX configuration shown inside the Https section, which is found earlier (above) in the current document :p The snippet found below is specific to serving static files but work in joint with the https configuration.

added to serve static files via NGINX.

.. code-block:: nginx

    map $sent_http_content_type $content_type_csp {
        default "";
        ~image/ "default-src 'none'";
    }

    server {
        # the rest of the configuration

        location @odoo {
            # copy-paste the content of the / location block
        }

        # Serve static files right away
        location ~ ^/[^/]+/static/.+$ {
            # root and try_files both depend on your addons paths
            root ...;
            try_files ... @odoo;
            expires 24h;
            add_header Content-Security-Policy $content_type_csp;
        }
    }

The actual ``root`` and ``try_files`` directives are dependant on your installation, specificaly on your :option:`--addons-path <odoo-bin --addons-path>`.

.. example::

   .. tabs::

      .. group-tab:: Debian package

>          Say Odoo has been installed via the **debian packages** for Community and Enterprise and

Reviewer: > Say Odoo has been installed

`Say Odoo was installed`

Same thing with the part about git

         the :option:`--addons-path <odoo-bin --addons-path>` is
         ``'/usr/lib/python3/dist-packages/odoo/addons'``.

         The ``root`` and ``try_files`` should be:

         .. code-block:: nginx

            root /usr/lib/python3/dist-packages/odoo/addons;
            try_files $uri @odoo;

      .. group-tab:: Git sources

         Say Odoo has been installed via the **sources**. The two git repositories for Community and

>          Enterprise have been cloned in :file:`/opt/odoo/community` and :file:`/opt/odoo/enterprise`

Reviewer: Maybe both parts should be in the same sentence.

```
Say Odoo was installed via the **source**, and both the Community and Enterprise repositories were cloned in ...
```

>          Say Odoo has been installed via the **sources**. The two git repositories for Community and
>          Enterprise have been cloned in :file:`/opt/odoo/community` and :file:`/opt/odoo/enterprise`

Reviewer: ```suggestion
         Say Odoo has been installed via the **sources**, and both the Community and Enterprise repositories were cloned in :file:`/opt/odoo/community` and :file:`/opt/odoo/enterprise`
```

Author: I reformulated the bit even further using this oxford coma in several other places for consistency
